test/abstract_syntax_tree.coffee

```diff
@@ -10,207 +10,595 @@ getExpressions = (code) ->
 
 getExpression = (code) -> getExpressions(code)[0]
 
+# Recursively compare all values of enumerable properties of `expected` with
+# those of `actual`. Use `looseArray` helper function to skip array length
+# comparison.
+
+deepStrictEqualExpectedProps = (actual, expected) ->
+  white = (text, values...) -> (text[i] + "#{reset}#{v}#{red}" for v, i in values).join('') + text[i]
+  eq actual.length, expected.length if expected instanceof Array and not expected.loose
+  for k , v of expected
+    if 'object' is typeof v
+      fail white"`actual` misses #{k} property." unless k of actual
+      deepStrictEqualExpectedProps actual[k], v
+    else
+      eq actual[k], v, white"Property #{k}: expected #{actual[k]} to equal #{v}"
+  actual
+
+# Flag array for loose comparision. See above code/comment.
+looseArray = (arr) ->
+  Object.defineProperty arr, 'loose',
+    value: yes
+    enumerable: no
+  arr
+
+testExpressions = (code, expected) ->
+  ast = getExpressions code
+  return console.log ast unless expected?
```

Comment:
I think I added these `console.log` lines as helpers for writing the tests, so I could figure out what `expected` should be. Now that all the tests are written we don’t need them anymore.

Reply:
I added those logs and would leave them there for future edits. They are very convenient.